### PR TITLE
Session Modal Service

### PR DESCRIPTION
**Pull request by Omicron7**
Initial implementation of a sessionModalService, signIn Modal and sign in button component.
Includes non-functional components for signUp and resetPassword.

@OzzieOrca @adammeyer I'd like some initial feedback on these while I work on implementing the signUp functionality.

### CODE REVIEW

File: src/common/components/resetPasswordModal/resetPasswordModal.tpl.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<div class="col-md-12 col-xs-12">
+  <p>Enter your registered email address below and we'll send a new password to that email.</p>
+  <form>
+    <div class="form-group">
+      <label for="email">Email</label>
+      <input type="email" class="form-control form-control-subtle" placeholder="Email">
+    </div>
+    <button type="submit" class="btn btn-block btn-primary">Reset My Password</button>
+  </form>
+  <p class="mt- text-center">Never mind
```
Comment (OzzieOrca) [inline src/common/components/resetPasswordModal/resetPasswordModal.tpl.html]: I feel like this needs a period or something after to separate it from the link. They are 2 different phrases.

File: src/common/components/signInButton/signInButton.component.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    'gettext',
+    template.name,
+    sessionService.name,
+    sessionModalService.name
+  ] )
+  .component( componentName, {
+    controller:  SignInButtonController,
+    templateUrl: template.name,
+    bindings:    {
+      pageReload: '<'
```
Comment (OzzieOrca) [inline src/common/components/signInButton/signInButton.component.js]: Is this going to be coming in from a variable in some outer scope or just hardcoded as a string on each btn instance such as `<sign-in-button page-reload="false">`? If it's the second case, I think using the `@` binding might be better.

File: src/common/components/resetPasswordModal/resetPasswordModal.tpl.html (hunk 1 above)
Comment (Omicron7) [inline src/common/components/resetPasswordModal/resetPasswordModal.tpl.html]: Thats copied directly from http://crugive.site/

File: src/common/components/signInButton/signInButton.component.js (hunk 2 above)
Comment (Omicron7) [inline src/common/components/signInButton/signInButton.component.js]: It's the later case, but '@' binding doesn't work with boolean values, changes them to strings.
The page-reload binding was something I added in the hope that other components become more session aware and don't need a complete page reload. We may never use this field, don't know yet.

File: src/common/components/resetPasswordModal/resetPasswordModal.tpl.html (hunk 1 above)
Comment (OzzieOrca) [inline src/common/components/resetPasswordModal/resetPasswordModal.tpl.html]: Gotcha but I still think that. Idk if my opinion has any weight.

**Comment by OzzieOrca** [comment]
I think this looks good. I like the component reuse and interaction and using the BehaviorSubject to broadcast changes in state to all the buttons.

File: src/common/components/signInButton/signInButton.component.js (hunk 2 above)
Comment (OzzieOrca) [inline src/common/components/signInButton/signInButton.component.js]: Alright ya. Whatever you think is best

**Comment by adammeyer** [comment]
👍

**Pull request merged**